backend/django_backend/bookings/views.py

```diff
@@ -76,14 +77,19 @@ def is_timeslot_available(self, service_id, service_provider_id, booking_date, s
 
 
 class DeleteBookingView(APIView):
-    def post(self, request, booking_id, *args, **kwargs):
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, booking_id, format=None):
         try:
             # Allow deletion if the request user is the owner or the service provider of the booking
             booking = Booking.objects.get(id=booking_id)
+            logging.info("Deleting Booking: %s", booking)
             if booking.user != request.user and booking.service_provider != request.user:
-                raise PermissionDenied("You do not have permission to delete this booking.")
+                logging.exception("Permission Denied: User does not have permission to delete the booking.")
+                return Response({"detail": "You do not have permission to delete this booking."}, status=status.HTTP_403_FORBIDDEN)
```

Comment:
Also, we should return this in the frontend when throwing error:

```
if(!response.ok) {
 const errorData = await response.json();
  throw new Error(errorData.detail || "Failed to create new post");
}
```

Reply:
I did not fix the handling of the error in the frontend since the changes in #44 will make them useless. Don't know how these changes actually got into this PR. My mistake!